addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2573,6 +2573,49 @@ var SnippetsMenu = Widget.extend({
             }
         });
 
+        //avoid dropzone collisions
+
+        function adjustMargin($zone, marginType, overlap, minMargin, maxMargin) {
+            let currentMargin = parseFloat($zone.css(marginType));
+            let newMargin = Math.min(Math.max(currentMargin + overlap, minMargin), maxMargin);
+            $zone.css(marginType, newMargin + "px");
+        }
+
+        // Sort the dropzones by their vertical position (top value)
+        $zones.sort(function(a, b) {
```

Comment:
You may need to configure a linter. By convention, there should be space after the `function` keyword. Or you could use an arrow function `(a, b) => {`.